// @mew/request/src/ms/mc.ts
import {IRes, mewMsPost, mewRequest, Request} from "../request";
import { isDebugMode } from "../demo";

export interface Role {
    grade?: number
    groupId?: number
    companyId?: number
    departmentId?: number
    teamId?: string
    username?: string
    status?: boolean
    description?: string
    name?: string
    id?: number

}

export interface AlipayTransferProperty{
    phoneNo?: string;
    firstname?: string;
    data: string;
    transAmount?: number;
}

export interface RoleSelectProperty extends Role {
    curPageNo?: number
    pageSize?: number
    startAt?: number
    endAt?: number
}

export interface RoleUpdateProperty extends Role {
    id: number

}

export interface RoleInsertProperty extends Role {
    grade: number
    status: boolean
    description: string
    name: string

}

export interface RoleDeleteProperty extends Role {
    id: number

}

// interface
// TODO 此处如果新增，需要修改 initDepartmentList 切记
export enum RoleName {
    unauthorized = 'unauthorized',
    company = 'company',
    company2 = 'company2',
    admin = 'admin',
    super = 'super',
    gtadmin = "gtadmin",
    gtdepart = 'gtdepart',
    gtdepart1 = 'gtdepart1'
}

export interface AuthProperty {
    username: string;
    password: string;
}

/// mc.manager
export interface Manager {
    id: number
    username: string
    password: string
    roleId: number
    groupId: number
    companyId: number
    departmentId: number
    teamId: string
    unionid: string
    status: boolean
    isSuper: boolean
    avatarUrl: string
    language: string
    createdAt: string
    updatedAt: string
}

export interface ManagerSelectProperty {
    id?: number
    companyId?: number
    groupId?: number
    departmentId?: number
    teamId?: string
    roleIdTest?: string
    username?: string
    password?: string
    roleId?: number
    unionid?: string
    status?: boolean
    language?: string
    isSuper?: boolean
    avatarUrl?: string
    pageSize?: number
    curPageNo?: number
    startAt?: Date
    endAt?: Date
}

export interface ManagerUpdateProperty {
    id: number
    username?: string
    password?: string
    roleId?: number
    unionid?: string
    language?: string
    status?: boolean
    isSuper?: boolean
    groupId?: number
    companyId?: number
    departmentId?: number
    teamId?: string
    avatarUrl?: string
}

export interface ManagerInsertProperty {
    id?: number
    username: string
    password: string
    roleId: number
    unionid?: string
    status: boolean
    isSuper: boolean
    groupId?: number
    companyId: number
    departmentId?: number
    language?: string
    teamId?: string
}

export interface ManagerDeleteProperty {
    id?: number
    username?: string
    password?: string
    roleId?: number
    unionid?: string
    language?: string
    status?: boolean
    isSuper?: boolean
    groupId?: number
    companyId?: number
    departmentId?: number
    teamId?: string
    avatarUrl?: string
}


/// mc.access
export interface Access {
    id: number
    moduleName: string
    actionName: string
    deviceTypeKind: string
    type: number
    url: string
    moduleId: number
    sort: number
    description: string
    config: string
    icon: string
    status: boolean
    createdAt: string
    updatedAt: string
    language?:string
    children: Access[]
}

export interface AccessSelectProperty {
    id?: number
    moduleName?: string
    actionName?: string
    deviceTypeKind?: string
    type?: number
    url?: string
    moduleId?: number
    sort?: number
    description?: string
    config?: string
    icon?: string
    status?: boolean
    pageSize?: number
    curPageNo?: number
    language?:string
    startAt?: Date
    endAt?: Date
}

export interface AccessUpdateProperty {
    id: number
    moduleName?: string
    actionName?: string
    deviceTypeKind?: string
    type?: number
    url?: string
    moduleId?: number
    sort?: number
    description?: string
    config?: string
    icon?: string
    status?: boolean
    language?:string
    kind?:string
}

export interface AccessInsertProperty {
    id?: number     // 可传id因为可能需要多环境公用配置
    moduleName: string
    actionName: string
    deviceTypeKind: string
    type: number
    url: string
    moduleId: number
    sort: number
    description?: string
    config?: string
    icon?: string
    status: boolean
    language?:string
    kind?:string
}

export interface AccessDeleteProperty {
    id: number
    moduleName?: string
    actionName?: string
    deviceTypeKind?: string
    type?: number
    url?: string
    moduleId?: number
    sort?: number
    description?: string
    config?: string
    icon?: string
    status?: boolean
    kind?:string
}

//role_access
export interface RoleAccess {
    id: number
    roleId: number
    accessId: number
    enabled: boolean
    createdAt: number
    updatedAt: number
}

export interface RoleAccessSelectProperty {
    id?: number
    inOrder?: boolean
    accessDetail?: boolean
    roleId?: number
    accessId?: number
    language?: string
    deviceTypeKind?: number
    enabled?: boolean
    pageSize?: number
    curPageNo?: number
    startAt?: number
    endAt?: number
}

export interface RoleAccessUpdateProperty {
    id: number
    roleId?: number
    accessId?: number
    enabled?: boolean
}

export interface RoleAccessDeleteProperty {
    id?: number
    roleId?: number
    accessId?: number
    enabled?: boolean
}

export interface RoleAccessInsertProperty {
    id?: number
    roleId: number
    accessId?: number
    enabled?: boolean
    accessIdArr?: number[]
}

export interface RoleAccessUpdateBatchProperty {
    roleId: number
    accessIdArr: number[]
}

// App
export interface App {
    id: string;
    appKey: string;
    appName: string
    appSecret: string;
    type: number;
    class: number;
    suiteTicket: string;
    suiteSecret: string;
    suiteId: string;
    suiteKey: string;
    agentId: string;
    authCorpId: string;
    status: boolean;
    createdAt: string
    updatedAt: string
}

export interface AppSelectProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
    curPageNo?: number;
    pageSize?: number;
    startAt?: number;
    endAt?: number;
}

export interface AppInsertProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
}

export interface AppUpdateProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
}

export interface AppDeleteProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
}

//auth
export interface AuthData {
    access_token: string
    expiration: number
    overdueDateTs: number
    overdueDate: string
}


export interface AccessTokenData {
    expireIn: number
    expireAt: number
    accessToken: string
}


export interface DingTalkUserInfoRes {
    errcode: number,
    errmsg: string,
    result: DingTalkUserInfo,
    request_id: string
}

export interface DingTalkUserInfo {
    device_id: string,
    name: string,
    sys: boolean,
    sys_level: number,
    unionid: string,
    userid: string
}

export interface getDingTalkUserInfoProperty {
    appId: string,
    code: string
}

export interface DingTalkMessageProperty {
    accessToken: string,
    content: string
}


// App
export interface App {
    id: string;
    appKey: string;
    appName: string
    appSecret: string;
    type: number;
    class: number;
    suiteTicket: string;
    suiteSecret: string;
    suiteId: string;
    suiteKey: string;
    agentId: string;
    authCorpId: string;
    status: boolean;
    createdAt: string
    updatedAt: string
}

export interface AppSelectProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
    curPageNo?: number;
    pageSize?: number;
    startAt?: number;
    endAt?: number;
}

export interface AppInsertProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
}

export interface AppUpdateProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
}

export interface AppDeleteProperty {
    id?: string;
    status?: boolean;
    secret?: string;
    class?: number;
    type?: number;
    appName?: string;
    suiteTicket?: string;
    suiteSecret?: string;
    suiteId?: string;
    suiteKey?: string;
    agentId?: string;
    authCorpId?: string;
    appid?: string;
}

export interface AccessTokenData {
    expireIn: number
    expireAt: number
    accessToken: string
}

export interface DingTalkUserInfoRes {
    errcode: number,
    errmsg: string,
    result: DingTalkUserInfo,
    request_id: string
}

export interface DingTalkUserInfo {
    device_id: string,
    name: string,
    sys: boolean,
    sys_level: number,
    unionid: string,
    userid: string
}

export class AppGetDingTalkUserInfo {
    appId?: string;
    code?: string;
}

export interface DingTalkMessageInterface {
    accessToken: string,
    content: string
}

export interface AppSendDingTalkMessage {
    appId: string;
    template_id: string;
    userid_list?: string;
    dept_id_list?: string;
    data: string;
}

export interface AppSendWxMessage {
    appId: string;
    template_id: string;
    touser: string;
    page: string;
    form_id: string;
    emphasis_keyword: string;
    data: any;
}

export interface getAppUserInfoData {
    session_key: string
    openid: string
    unionid: string
}

export interface GetAccessTokenProperty {
    appId?: string;
}

export interface AuthProperty {
    username: string;
    password: string;
}

//auth
export interface AuthData {
    access_token: string
    expiration: number
    overdueDateTs: number
    overdueDate: string
}

export function Mc(service: any, isDebugMode: boolean = false,isAesToStr = false ) {
    return {
        role: new Request<Role, RoleInsertProperty, RoleDeleteProperty, RoleUpdateProperty, RoleSelectProperty>('/mc/role', service, isDebugMode,isAesToStr),
        // class
        manager: new Request<Manager, ManagerInsertProperty, ManagerDeleteProperty, ManagerUpdateProperty, ManagerSelectProperty>('/mc/manager', service, isDebugMode,isAesToStr),
        access: new Request<Access, AccessInsertProperty, AccessDeleteProperty, AccessUpdateProperty, AccessSelectProperty>('/mc/access', service, isDebugMode,isAesToStr),
        role_access: new Request<RoleAccess, RoleAccessInsertProperty, RoleAccessDeleteProperty, RoleAccessUpdateProperty, RoleAccessSelectProperty>('/mc/role_access', service, isDebugMode,isAesToStr),
        app: new Request<App, AppInsertProperty, AppDeleteProperty, AppUpdateProperty, AppSelectProperty>('/mc/app', service, isDebugMode,isAesToStr),
        // function
        getAuth: (json: AuthProperty) => mewRequest('/auth', json, 'POST', service,isDebugMode,isAesToStr) as Promise<{ code: number, msg: string, data: AuthData }>,
        getVersion: () => mewRequest('/mc/select/version', {}, 'POST',  service,isDebugMode,isAesToStr),
        getAccessToken: async (appid: string): Promise<{ code, msg, data: AccessTokenData }> => {
            return await mewRequest('/mc/app/getAccessToken', {appid}, 'POST',  service,isDebugMode,isAesToStr)
        },
        // getCorpAccessToken: async (json:GetAccessTokenProperty): Promise<{ code, msg, data: AccessTokenData }> => {
        //     return await mewRequest( '/mc/app/getCorpAccessToken', json,'POST', service,isDebugMode)
        // },
        /**
         * 废弃，使用 getAppUserInfo
         * @param json
         */
        getDingTalkUserInfo: async (json: getDingTalkUserInfoProperty): Promise<DingTalkUserInfoRes> => {
            return await mewRequest('/mc/app/getDingTalkUserInfo', json, 'POST',  service,isDebugMode,isAesToStr)
        },
        getAppUserInfo: async (json: AppGetDingTalkUserInfo): Promise<IRes<getAppUserInfoData>> => {
            return await mewRequest('/mc/app/getUserInfo', json, 'POST',  service,isDebugMode,isAesToStr)
        },
        getUserPhoneNo: async (json: AppGetDingTalkUserInfo): Promise<IRes<getAppUserInfoData>> => {
            return await mewRequest('/mc/app/getUserPhoneNo', json, 'POST',  service,isDebugMode,isAesToStr)
        },
        sendDingTalkMessage: async (json: AppSendDingTalkMessage) => {
            return await mewRequest('/mc/app/sendDingTalkMessage', json, 'POST', service,isDebugMode,isAesToStr)
        },
        sendWeiXinMessage: async (json: AppSendWxMessage) => {
            return await mewRequest('/mc/app/sendWxMessage', json, 'POST',  service,isDebugMode,isAesToStr)
        },
        getAccessByRole: async (json: RoleAccessSelectProperty) => {
            return await mewRequest('/mc/role_access/getAccessByRole', json, 'POST', service,isDebugMode,isAesToStr)
        },
        // 批量修改角色权限
        updateRoleAccessBatch: async (json: RoleAccessUpdateBatchProperty): Promise<IRes> => {
            return await mewMsPost(`/mc/role_access/update/batch`, json,  service,isDebugMode,isAesToStr)
        },
        // access直接导入
        accessDirectExcelSave: async (json: any): Promise<IRes> => {
            return await mewMsPost(`/mc/access/directExcelSave`, json, service, isDebugMode, isAesToStr)
        },
        alipayTransfer: (json: AlipayTransferProperty) => mewRequest('/mc/app/alipayTransfer', json, 'POST', service,isDebugMode,isAesToStr),
    }

}

export function getAuth(json: AuthProperty, service) {
    return mewRequest('/auth', json, 'POST',  service) as Promise<{ code: number, msg: string, data: AuthData }>
}
