// @mew/request/dist/ms/dashboard.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Dashboard = void 0;
var tslib_1 = require("tslib");
var request_1 = require("../request");
function Dashboard(service, isDebugMode, isAesToStr) {
    var _this = this;
    if (isDebugMode === void 0) { isDebugMode = false; }
    if (isAesToStr === void 0) { isAesToStr = false; }
    return {
        // function
        cacheGetData: function () { return tslib_1.__awaiter(_this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, request_1.mewMsPost('/cache/get', { key: "localStatus", index: 0 }, service, isDebugMode, isAesToStr)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); },
        /**
         * 获取工具借还情况,根据天
         */
        toolSituation: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewMsPost('/statistics/logByDay', json, service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 获取工具使用前十
         * 工具使用排行榜top10
         * /log/getTopTenUsageTimes
         */
        toolTopTen: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewMsPost('/statistics/topTenUsageTimes', json, service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 获取工具使用后十
         * 工具使用排行榜top10
         * /log/getTopTenUsageTimes
         */
        toolBottomTen: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewMsPost('/statistics/bottomTenUsageTimes', json, service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 部门使用情况
         */
        departmentFrequency: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewMsPost('/statistics/departmentFrequency', json, service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 班组使用情况
         */
        teamFrequency: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewMsPost('/statistics/departmentFrequency', json, service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 班组使用情况
         */
        topTenUsers: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewMsPost('/statistics/topTenUsers', json, service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 设备状态
         */
        deviceStatus: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/deviceStatus', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 基础数据,不传数据,默认最近30天
         * 1.回收垃圾总量
         * 2.发放积分
         * 3.投递次数
         * 4.清运次数
         * 5.小区数量
         * 6.用户数量
         * 7.今日回收重量
         * 8.今日使用频次
         */
        baseData: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/baseData', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 垃圾分类投递重量数据
         */
        garbageWightData: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/garbageWightData', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 垃圾比例
         */
        garbageWeightProportion: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/garbageWeightProportion', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 垃圾四分类
         */
        garbageWeightPrimary: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/garbageWeightPrimary', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 垃圾比例,按照设备排名
         */
        garbageWeightOrderByDevice: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/garbageWeightOrderByDevice', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 大屏数据
         */
        bmData: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/bmData', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         *  获取缓存
         */
        cacheGet: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/cache/get', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         *  获取缓存
         */
        cacheGetByKey: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/cache/getByKey', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         *  获取版本
         */
        getVersion: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/version', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         *   设置缓存
         */
        cacheSet: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/cache/set', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        },
        /**
         * 大屏数据
         */
        lackToolList: function (json) {
            return tslib_1.__awaiter(this, void 0, void 0, function () {
                return tslib_1.__generator(this, function (_a) {
                    switch (_a.label) {
                        case 0: return [4 /*yield*/, request_1.mewRequest('/statistics/lackToolList', json, 'POST', service, isDebugMode, isAesToStr)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            });
        }
    };
}
exports.Dashboard = Dashboard;
